pallets/gear/src/runtime_api.rs

```diff
@@ -211,8 +211,14 @@ where
             for note in journal {
                 core_processor::handle_journal(vec![note.clone()], &mut ext_manager);
 
-                if let Some(remaining_gas) = get_main_limit() {
-                    min_limit = min_limit.max(initial_gas.saturating_sub(remaining_gas));
+                match get_main_limit() {
+                    Some(remaining_gas) => {
+                        min_limit = min_limit.max(initial_gas.saturating_sub(remaining_gas))
+                    }
+                    None if matches!(note, JournalNote::WaitDispatch { .. }) => {
+                        min_limit = initial_gas
```

Comment:
it's not always "infinite" gr_wait()

Reply:
then it will go to the `Some(...)` branch since there will be some gas value in the node

Reply:
Yes, but do we have problems with `gr_wait_for(dur)` and `gr_wait_up_to(dur)`? They should not consume all available gas because of limited duration.

Reply:
Frankly speaking bounded syscalls can have an enough big duration argument so additional check wouldn't be redundant